src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/BodyModelBinder.cs

```diff
@@ -65,9 +65,6 @@ public Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContex
                 throw new ArgumentNullException(nameof(bindingContext));
             }
 
-            // For compatibility with MVC 5.0 for top level object we want to consider an empty key instead of
-            // the parameter name/a custom name. In all other cases (like when binding body to a property) we
-            // consider the entire ModelName as a prefix.
```

Comment:
The following code in `CompositeModelBinder` and the fact `ModelName` is initially equal to `FieldName` implies the opposite.
``` c#
    if (bindingSource != null && bindingSource.IsGreedy)
    {
        newBindingContext.ModelName = oldBindingContext.ModelName;
    }
```

But if I'm reading that wrong, how 'bout
``` c#
    var modelBindingKey = bindingContext.ModelName;
    if (string.IsNullOrEmpty(modelBindingKey)
    {
        modelBindingKey = bindingContext.FieldName;
    }
```